gen/nvml/lib.go

```diff
@@ -25,7 +30,64 @@ const (
 	nvmlLibraryLoadFlags = dl.RTLD_LAZY | dl.RTLD_GLOBAL
 )
 
-var nvml *dl.DynamicLibrary
+type nvmlLib struct {
+	mu  sync.Mutex
+	lib *dl.DynamicLibrary
+}
+
+// nvml represents the global nvml library.
+var nvml nvmlLib
+
+// Open ensures that the nvml library is initialized.
+// If it is not initialized, the library is opened using dlopen and the versioned
+// symbols are updated.
+func (l *nvmlLib) Open() error {
```

Comment:
Given this comment, should be be called `init` instead of `open`? Or should the comment be updated? 

Probably just the comment updated to:
```
// Open ensures that the nvml library is open and initialized with the proper versioned symbols.
// Multiple calls to an already opened nvml library will return without error.
```

Reply:
Yes, I think `init` or `initLibrary` is better here. Note that I have created #86 which already includes this (or something similar) for the `Lookup` changes we discussed.

Reply:
I elected to go with `library.load()` here. See the changes in #86.